step-framework-parent/step-framework-collections-postgresql/src/main/java/step/core/collections/postgresql/PostgreSQLCollection.java

```diff
@@ -322,15 +329,24 @@ public void createOrUpdateCompoundIndex(String... fields) {
 
 	@Override
 	public void createOrUpdateCompoundIndex(Map<String, Integer> fields) {
+		LinkedHashSet<IndexField> indexFields = fields.entrySet().stream()
```

Comment:
This is one of these cases where the resulting index field order is non-deterministic. Ideally, this should be deprecated and removed in the future, but for now it's ok according to the discussion.